src/codemagic/tools/app_store_connect/actions/fetch_signing_files_action.py

```diff
@@ -0,0 +1,237 @@
+        CertificateArgument.P12_CONTAINER_PASSWORD,
+        ProfileArgument.PROFILE_TYPE,
+        BundleIdArgument.IDENTIFIER_STRICT_MATCH,
+        CommonArgument.CREATE_RESOURCE,
+    )
+    def fetch_signing_files(
+        self,
+        bundle_id_identifier: str,
+        certificate_key: Optional[Union[PrivateKey, Types.CertificateKeyArgument]] = None,
+        certificate_key_password: Optional[Types.CertificateKeyPasswordArgument] = None,
+        p12_container_password: str = "",
+        platform: BundleIdPlatform = BundleIdPlatform.IOS,
+        profile_type: ProfileType = ProfileType.IOS_APP_DEVELOPMENT,
+        bundle_id_identifier_strict_match: bool = False,
+        create_resource: bool = False,
+    ) -> Tuple[List[Profile], List[SigningCertificate]]:
+        """
+        Fetch provisioning profiles and code signing certificates
+        for Bundle ID with given identifier
+        """
+
+        private_key = self._get_certificate_key(certificate_key, certificate_key_password)
+        if private_key is None:
+            raise AppStoreConnectError(f"Cannot save {SigningCertificate.s} without certificate private key")
+
+        bundle_ids = self._get_or_create_bundle_ids(
+            bundle_id_identifier,
+            platform,
+            create_resource,
+            bundle_id_identifier_strict_match,
+        )
+        self.echo("")
+
+        certificates = self._get_or_create_certificates(
+            profile_type,
+            certificate_key,
+            certificate_key_password,
+            create_resource,
+        )
+        self.echo("")
+
+        profiles = self._get_or_create_profiles(
+            bundle_ids,
+            certificates,
+            profile_type,
+            create_resource,
+            platform,
+        )
+        self.echo("")
+
+        self._save_certificates(certificates, private_key, p12_container_password)
+        self._save_profiles(profiles)
+        return profiles, certificates
+
+    def _get_or_create_bundle_ids(
+        self,
+        bundle_id_identifier: str,
+        platform: BundleIdPlatform,
+        create_resource: bool,
+        strict_match: bool,
+    ) -> List[BundleId]:
+        bundle_ids = self.list_bundle_ids(
+            bundle_id_identifier,
+            platform=platform,
+            bundle_id_identifier_strict_match=strict_match,
+            should_print=False,
+        )
+        if not bundle_ids:
+            if not create_resource:
+                raise AppStoreConnectError(f"Did not find {BundleId.s} with identifier {bundle_id_identifier}")
+            bundle_ids.append(self.create_bundle_id(bundle_id_identifier, platform=platform, should_print=False))
+        else:
+            for bundle_id in bundle_ids:
+                self.logger.info(f"- {bundle_id.attributes.name} {bundle_id.attributes.identifier} ({bundle_id.id})")
+        return bundle_ids
+
+    def _get_or_create_certificates(
```

Comment:
Used to be defined in https://github.com/codemagic-ci-cd/cli-tools/blob/3670fa624997f17cb437bc88198f7bdee3594f35/src/codemagic/tools/app_store_connect.py#L907-L945